text/python/0000-dependencies-update.md

```diff
@@ -0,0 +1,97 @@
+# Decide which Python dependencies will be Paketo-hosted
+
+## Proposal
+
+Currently four Python buildpacks have Paketo-hosted dependencies.
+Only two Paketo-hosted dependencies should be kept, and two should be removed.
+
+| Buildpack                                                                        | Action |
+|----------------------------------------------------------------------------------|--------|
+| [cpython](https://github.com/paketo-buildpacks/cpython/blob/main/buildpack.toml) | Keep   |
+| [poetry](https://github.com/paketo-buildpacks/poetry/blob/main/buildpack.toml)   | Remove |
+| [pip](https://github.com/paketo-buildpacks/pip/blob/main/buildpack.toml)         | Keep   |
+| [pipenv](https://github.com/paketo-buildpacks/pipenv/blob/main/buildpack.toml)   | Remove |
+
+### Offline not supported
+
+Note that the Python language family does not support a fully-offline case, since several of its components require internet connectivity to work.
+These include at least the following: 
+
+* pipenv: [#limitations](https://github.com/paketo-buildpacks/pipenv#limitations)
+* poetry: [#known-issues-and-limitations](https://github.com/paketo-buildpacks/poetry#known-issues-and-limitations)
+
+## Rationale
+
+### CPython
+
+Keep this as a Paketo-hosted dependency.
+
+Currently this uses `dep-server`, `binary-builder`, and `buildpacks-ci` to build, but language family maintainers
+will transition this to the new Github Action workflow once that has been approved.
```

Comment:
The rationale behind why we need to continue hosting this ourselves is not clear. What's the problem with using the version directly from upstream?

Reply:
What's going on in the compilation code that's needed?

Reply:
There are currently [quite a lot of steps](https://github.com/cloudfoundry/buildpacks-ci/blob/ff0027d31d6749d67670b83a7b2da500d992c435/tasks/build-binary-new/builder.rb#L248) to build cpython. We could explore moving all these steps to the buildpack but I don't think it's a trivial change.

Reply:
The compilation process involves installing system packages, so this RFC proposes keeping `CPython` as a Paketo-hosted dependency built outside of the buildpack.